addons/mail/static/src/core/common/partner_suggestion_compare_fn.js

```diff
@@ -0,0 +1,141 @@
+/* @odoo-module */
+
+import { cleanTerm } from "@mail/utils/common/format";
+import { registry } from "@web/core/registry";
+
+/**
+ * Registry of functions to sort partner suggestions.
+ * The expected value is an object of the following form:
+ *    { sequence: number, fn: (partner1, partner2, { searchTerms: string, thread?: Thread }) => number|undefined }
+ * Note that the compare functions will be call in the context of the
+ * caller (i.e. the SuggestionService).
```

Comment:
Is it really necessary? Can't we pass whatever is used as param? It's only used for recent chats?

Reply:
I passed the `env`. `PersonaService` seems too specific to be useful and it would not be surprising that other compare functions need it.